File: day:48/isIsomorphic.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        #base case 
        if(len(s) != len(t)): 
            return False;
    
        hash = {}; 
    
        for i in range(len(s)):
            [x, y] = [s[i], t[i]]
            if(x in hash):
                if(hash[x] != y):
                    return False;
            else:
                #x is new
                #check y is also new
                if(y in hash.values()):
                    return False;
            hash[x] = y;
                    
        return True;

File: day:48/isIsomorphic.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isIsomorphic = function(s, t) {
    // base case 
    if(s.length !== t.length) return false;
    
    let hash = {}; // tracks whether the character has been parsed
    
    for(let i = 0; i < s.length; i++) {
        let [x, y] = [s[i], t[i]];
        if(x in hash){
            if(hash[x] !== y) {
                return false;
            }
        }
        else {
            // x is new
            // check y is also new
            if(Object.values(hash).includes(y)) {
                return false;
            }
            hash[x] = y;
        }
    }
    return true;
    
};
